src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -238,26 +222,37 @@ private static DetailAST createCommentAstFromToken(Token token) {
      * @param token to create the AST
      * @return DetailAST with SINGLE_LINE_COMMENT type
      */
-    private static DetailAST createSlCommentNode(Token token) {
+    private static DetailAST createSlCommentNode(CommonToken token) {
         final DetailAstImpl slComment = new DetailAstImpl();
         slComment.setType(TokenTypes.SINGLE_LINE_COMMENT);
         slComment.setText("//");
 
-        // column counting begins from 0
-        slComment.setColumnNo(token.getColumn() - 1);
+        slComment.setColumnNo(token.getCharPositionInLine());
         slComment.setLineNo(token.getLine());
 
         final DetailAstImpl slCommentContent = new DetailAstImpl();
         slCommentContent.setType(TokenTypes.COMMENT_CONTENT);
 
-        // column counting begins from 0
         // plus length of '//'
-        slCommentContent.setColumnNo(token.getColumn() - 1 + 2);
+        slCommentContent.setColumnNo(token.getCharPositionInLine() + 2);
         slCommentContent.setLineNo(token.getLine());
         slCommentContent.setText(token.getText());
 
         slComment.addChild(slCommentContent);
         return slComment;
     }
 
+    /**
+     * Custom error listener to provide detailed exception message.
+     */
+    private static class CheckstyleErrorListener extends BaseErrorListener {
+
+        @Override
+        public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol,
+                                int line, int charPositionInLine,
+                                String msg, RecognitionException ex) {
+            final String message = line + ":" + charPositionInLine + ": " + msg;
+            throw new IllegalStateException(message, ex.getCause());
```

Comment:
`throw new IllegalStateException(message, ex);`

if exception happens, never hide any part of it.

Reply:
Excellent  point. Done.